### ISSUE TITLE
Add Bootstrap alternatives

### ISSUE THREAD

**Issue by Awk34** [DaftMonk/generator-angular-fullstack#999]
It would be great for users to be able to choose between different UI frameworks, such as [Foundation](http://foundation.zurb.com/), [Semantic UI](http://semantic-ui.com/), [Refills](http://refills.bourbon.io/), [Angular Material](https://material.angularjs.org/latest/)....

**Comment by remicastaing** [comment]
It would be awesome but in the same time a nightmare to maintain...

Don't forget the upcoming [Material Design Lite](https://developers.google.com/web/tools/material-design-lite/). ;)

**Comment by Awk34** [comment]
I don't agree that it would be a nightmare to maintain. I would most likely split them out into separate template files with different lib inclusions.

MD lite looks cool, is it Polymer 1.0 stuff?

**Comment by remicastaing** [comment]
I'm not sure. I didn't find any useful info about MDL.

**Comment by almandot** [comment]
How "all inclusive" does this project hope to be?

As a pretty much noob who used this to start off my school project last year it seemed pretty trivial to choose not to include bootstrap and just bower install foundation on my own along with things like angular-foundation or whatever other solutions I needed (I think the biggest challenge was having to figure out that I needed to set htmlmin's removeRedundantAttributes to false). It wasn't like css styling was rooted into anything essential in how the app was set up... thankfully not.

**Comment by Awk34** [comment]
I'd need to look into how complex modifying the generator templates would be to allow for other UI frameworks. Of course just including different Bower dependencies would be easy as pie.

**Comment by josephmilla** [comment]
+1 for this

**Comment by DanMMX** [comment]
I'm afraid for some of them it would need to be supported Sass in order to compile it cleanly as using Bootstrap right now (including `bootstrap.less` into `app.less`). Personally I wouldn't add Sass only for including any of these, but would thing it looks 'dirty' including raw css into the `app.less` file.

**Comment by remicastaing** [comment]
You can leave MDL aside for some time. As stated in https://github.com/google/material-design-lite/issues/1175, SPA aren't fully supported today:
>There’s a lot of work to be done, but once we have fulfilled our goals sufficiently, there’s a good chance we’ll promote SPAs to first-class citizens. But that’s something the probably won’t (and can’t) happen before v2.

I tried it and encountered some weird weird behavior: forms working the first time and by reloading doesn't working anymore.

**Comment by pablodgonzalez** [comment]
Hi guys! What is the status of this feature?

**Comment by derMani** [comment]
Has there been any development here? 

Is this repository known? 

https://github.com/sincraianul/generator-angular-material-fullstack

This guy used an older fork of the generator and made angular material templates.

Works pretty well, unfortunately it's not very well maintained.

**Comment by Koslun** [comment]
Don't think there's been in any development and think you'll generally be better off using a newer version of this generator and then integrating Angular Material in after that. Think it would be a slight pain to get Angular Material now as we're preparing to move to Angular 2 as Angular Material depends on Angular 1. Think it instead rather makes more sense to just move to Angular 2 faster and once we have all those optional features in Angular 2 ready we can maybe instead look at Angular Material 2, which might very well be in beta by then.

Mainly because I think just getting an Angular 2 stack is harder and more urgent than any extra UI library.

**Comment by Awk34** [comment]
@derMani No..

Yes, I've seen that repo before.

One of the things that made maintaining this project so hard in the past was that there were separate files for JS & CoffeeScript. Now we only have one template file for both JS & TypeScript, and maintaining the JS is a breeze compared to having to update two template files. I plan to eventually do the same for the templates and CSS.

So how would we set up the ability to use these different CSS frameworks without introducing more maintainabliity nightmares? That's the real question that has to be answered. One CSS framework that would be a breeze to add to the current stack would be something like [bootstrap-material-design](https://fezvrasta.github.io/bootstrap-material-design/), since there would be minimal changes on top of the current CSS classes/etc.

As for Angular 2, yes, once we move to that route, we'll want to add material2 as an option. I see one of the easiest transitions there in changing which components framework to use (`<md-modal [propx]="data">` vs `<another-modal [propy]="data">`)

**Comment by macneib** [comment]
Been using ng2 material for a few weeks and it's painless.  It's still very alpha, meaning there's not much there, but what is there is first class.  However, I think it would be a lot of wasted effort to bring ng1 material into the mix at this time.
